include/astarte_device_sdk/device.h

```diff
@@ -216,9 +225,15 @@ astarte_result_t astarte_device_connect(astarte_device_handle_t device);
  * @note It will be possible to re-connect the device after disconnection.
  *
  * @param[in] device Device instance to be disconnected.
+ * @param[in] timeout Timeout for this function when @p force is false.
+ * @param[in] force Force the disconnection. When false this function will be blocking and wait
+ * untill all QoS 1 or larger messages have been successfully transmitted, or a timeout has been
+ * reached. When true this function will not wait for any message to be correctly delivered and
+ * force a device disconnection.
  * @return ASTARTE_RESULT_OK if successful, otherwise an error code.
  */
-astarte_result_t astarte_device_disconnect(astarte_device_handle_t device);
+astarte_result_t astarte_device_disconnect(
+    astarte_device_handle_t device, k_timeout_t timeout, bool force);
```

Comment:
Since If It passed a timeout the function has a different behavior, What do you think to separate in two different function?
``` c
astarte_result_t astarte_device_disconnect(
    astarte_device_handle_t device, bool force);
astarte_result_t astarte_device_disconnect_with_timeout(
    astarte_device_handle_t device, k_timeout_t timeout);
```

Reply:
The `force` flag would be unnecessary in the first function, making a forced disconnection the default.
I think the disconnect function should not default to disconnecting the device disregarding pending QoS 1 and 2 messages.
I prefer having a single function with a force argument to clarify what the function does.
Another option would be to keep the force argument in the disconnect function and return an error when force is not set and there are pending messages.

Reply:
I would split them in a different way to avoid having the force boolean. The inner workings should be the same but without the bool parameter.
```suggestion
// does not block and disconnects the client imediately
astarte_result_t astarte_device_force_disconnect(
    astarte_device_handle_t device);
// wait that messages get sent or for the specified timeout 
astarte_result_t astarte_device_disconnect(
    astarte_device_handle_t device, k_timeout_t timeout);
```
This way we avoid having weird situations where a user could define a timeout even if it is unused because force is true.

Reply:
I think this could be a good middle ground, clear enough for the user to understand while still separating the functions.